### DISCUSSION TITLE
How to allow graceful control+c?

### DISCUSSION THREAD

**Discussion by IdanDavidi** [piotrmurach/tty-prompt#171] [Q&A] [answered]
Hey guys,
Is it possible to allow graceful control+c instead of getting this message?
![image](https://user-images.githubusercontent.com/9549288/126336509-2cce809d-d566-442f-910f-a2b9a759aba3.png)

I came up with this solution but it exits the console:
```ruby
prompt = TTY::Prompt.new(interrupt: ->() { puts "\nAborted.".yellow; exit(0) })
```

So I guess I should replace the `exit(0)` with something else?
Thanks.

**Comment by piotrmurach** [comment]
Hi Idan 👋 

> Is it possible to allow graceful control+c instead of getting this message?

It would be helpful if you could be more specific by what you mean by 'graceful control+c'. What do you want your CLI to do when a user presses `Ctrl+C`? You have full control over what happens when `Ctrl+C` is pressed and it's up to you to decide.

For example, it is quite common for programs to stop running when `Ctrl+C` is pressed. Often the script returns a non-zero code, in this case, code `130`. You may consider using [tty-exit](https://github.com/piotrmurach/tty-exit) for this like so:

```ruby
prompt  = TTY::Prompt.new(interrupt: -> { TTY::Exit.exit_with(:interrupt) })
```

Then applications that rely on the standard exit code can respond appropriately.

Another option, if you do not wish to exit from your script is to, for example, capture the `TTY::Reader::InputInterrupt` error and do whatever you wish when the input gets interrupted.

**Reply by IdanDavidi** [reply]
@piotrmurach sorry for not being specific enough.
By default, when a user presses `Ctrl + C`, an error is thrown and printed to the console:
![controlc](https://user-images.githubusercontent.com/9549288/126443395-8871c3ef-8523-40b1-8781-2e329162aff6.gif)

What I wish to achieve is not to show this error in the console. 
I know that I can rescue this specific error, but is it possible to achieve that using the `interrupt` option?

**Comment by garrettd714** [comment, marked as answer]
I added a `rescue` clause on #execute in command file

```ruby
rescue TTY::Reader::InputInterrupt
  puts pastel.red('Cancelled input. Data not saved')
end
```
![Screen Shot 2021-11-18 at 10 58 34 AM](https://user-images.githubusercontent.com/243288/142479436-d948d28c-5445-4a34-a7cf-679cb236117d.png)

**Reply by garrettd714** [reply]
Sorry, didn't see the last line in Question...
